pkg/internal/util/kubeconfig/encode_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconfig
+
+import (
+	"testing"
+
+	"sigs.k8s.io/kind/pkg/internal/util/assert"
+)
+
+func TestEncodeRoundtrip(t *testing.T) {
```

Comment:
can we add a test failing?

Reply:
not trivially, it only fails if it cannot marshal the type so ...

Reply:
FYI we have 41.5% statements covered in this package, the rest are mostly error cases that aren't really triggerable like this, or file system interactions (and somewhat unit tests then, but I will probably still add some)